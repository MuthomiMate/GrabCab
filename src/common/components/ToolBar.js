import React, { Component } from "react";
import { View, Text, StyleSheet, StatusBar } from "react-native";
import { Icon } from "react-native-elements"

const ToolBar = ({iconName, iconSize, iconType, iconColor, text, leftIcon, onleftIconPress, onRightIconClick, textColor }) => {
    return (
        <View style={styles.container}>
        <StatusBar backgroundColor="#2EBD6B" barStyle="light-content" />
            <View style={styles.detailsView} >
                <View>
                    {iconName ? <Icon onPress={ onRightIconClick } name={iconName} size={iconSize} type={iconType} color={iconColor} />: null }
                </View>
                <View>
                    {text ? <Text style={{color: textColor}} onPress={ onRightIconClick }>{text}</Text>: null}
                </View>
                <View>
                    {leftIcon ? <Icon name={iconName} size={iconSize} type={iconType} color={iconColor} /> : null }
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        width: "100%",
        backgroundColor: "#0A0F10",
        height: "30%"
    },
    detailsView: {
        flexDirection: "row",
        justifyContent: "space-between",
        top: "10%"
    }
})
export default ToolBar;