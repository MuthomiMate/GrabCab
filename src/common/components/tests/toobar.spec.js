import React from "react";
import { shallow } from "enzyme";
import ToolBar from "../ToolBar";

describe("<ToolBar />", () => {
    const testProps = {
        iconName:"menu",
        iconSize:25,
        iconType:"entypo",
        iconColor:"#FFFFFF",
        text:"DashBoard",
        textColor:"#FFFFFF"
    }

    it("renders without crashing", () => {
        const wrapper = shallow(<ToolBar {...testProps}/>);
        expect(wrapper).toMatchSnapshot();
    })
})