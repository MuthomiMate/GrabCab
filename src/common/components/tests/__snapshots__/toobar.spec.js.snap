// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolBar /> renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0A0F10",
      "height": "30%",
      "width": "100%",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#2EBD6B"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "top": "10%",
      }
    }
  >
    <View>
      <Themed.Icon
        color="#FFFFFF"
        name="menu"
        size={25}
        type="entypo"
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        DashBoard
      </Text>
    </View>
    <View />
  </View>
</View>
`;
